### PR TITLE
Added `ashist()` method to stats

### LINKED ISSUE #117: Add features to `degree_histogram`
I wanted to easily visualize the degree/edge size distribution and the way that I found easiest to get both was the following:
```
degrees = dict(H.degree())
edge_sizes = dict(H.edge_size())

d, count_d = np.unique(list(degrees.values()), return_counts=True)
p_d = count_d/len(degrees)
e, count_e = np.unique(list(edge_sizes.values()), return_counts=True)
p_e = count_e/len(edge_sizes)
```

Currently, there is a way to do some of this with `degree_histogram`, but I would like to be able to a) specify either edge sizes or degrees and b) specify whether it is normalized. And perhaps we should return numpy arrays instead of lists.

### PR DESCRIPTION
**Pull request by nwlandry**
Fixes #117. Added a method to compute the distribution of an array of numbers. Design decisions attempted to roughly follow the discussion in #117.

### CODE REVIEW

**Comment by maximelucas** [comment]
Great, I've been hoping for this feature for so long!

Looks good. Two thoughts:
- you return ndarrays (of same size), and not df as Leo suggested, right? I'm fine with both, but better check with @leotrs . Maybe one of the reasons was that it's more easily plottable?
- when I read `asdist` I tend to think of "distance" rather than "distribution". Do you think `ashist()` could make sense?

**Comment by nwlandry** [comment]
Good call. There are definitely some discrepancies between what we decided in that issue vs. what I did. My biggest takeaways are (1) call this `ashist()` instead of `dist`, (2) allow users to get bin edges as well as centers, and (3) return as Pandas instead of two numpy arrays. From looking back at that conversation, I think that implementing both `hist` and `dist` seems like overkill but there are some nice suggestions that I will spend some time incorporating.

File: xgi/utils/utilities.py [hunk 1]
```diff
@@ -415,3 +417,57 @@ def convert_labels_to_integers(net, label_attribute="label"):
...
+            bins = np.logspace(lower_bound, upper_bound, bins + 1, base=10)
+        else:
+            bins = np.linspace(lower_bound, upper_bound, bins + 1)
+    elif not isinstance(bins, (list, np.ndarray)):
+        raise XGIError("Bins must be an integer, a list, or a numpy array.")
+
+    # Then we can compute the histogram using numpy
+    y, __ = np.histogram(vals, bins=bins, density=density)
+    # Now, we need to compute for each y the bin centers
+    x = bins[1:] - np.diff(bins) / 2.0
```
[the remark this answers is not in the document]
Comment (nwlandry) [reply xgi/utils/utilities.py:471]: Working on this right now!

**Comment by nwlandry** [comment]
@maximelucas @leotrs I think I addressed your comments - let me know if there's anything else I can address!

**Comment by maximelucas** [comment]
Thanks Nich, functionality looks good to me. 

Two questions:
- it looks like only `xgi.hist()` is tested, should we also test say `H.nodes.degree.ashist()`?
- Unsure but might still need to be added to the docs?

**Comment by nwlandry** [comment]
> Thanks Nich, functionality looks good to me.
> 
> Two questions:
> 
> * it looks like only `xgi.hist()` is tested, should we also test say `H.nodes.degree.ashist()`?
> * Unsure but might still need to be added to the docs?

I opted to only test the xgi.hist function because the tests for the stats package are a bit of a mess right now IMO and the stats functions directly call xgi.hist() so all the functionality is tests. If it's okay with you, I will make an issue to reorganize the stats tests (we can discuss the best way to do this) so that it's more organized.

Good catch on the docs - fixed now.

**Review by maximelucas — APPROVED** [review]

**Pull request merged**